src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java

```diff
@@ -7,50 +7,158 @@
 import javax.annotation.CheckForNull;
 import javax.annotation.Nonnull;
 import jenkins.model.FingerprintFacet;
+import org.apache.commons.collections.CollectionUtils;
 
 /**
  * Entry point into fingerprint related functionalities in Docker.
+ * This class provide basic methods for both images and containers
  */
 public class DockerFingerprints {
+    
+    private static final Logger LOGGER = Logger.getLogger(DockerFingerprints.class.getName());
+    
     private DockerFingerprints() {} // no instantiation
  
     /**
-     * Gets a fingerprint hash for Docker image ID.
+     * Gets a fingerprint hash for Docker ID (image or container).
      * This method calculates image hash without retrieving a fingerprint by 
      * {@link DockerFingerprints#of(java.lang.String)}, which may be a high-cost call.
      * 
-     * @param imageId Docker image ID.
-     *      Only 64-char full image IDs are supported.
+     * @param id Docker ID (image or container).
+     *      Only 64-char full IDs are supported.
      * @return 32-char fingerprint hash
-     * @throws IllegalArgumentException Invalid image ID
+     * @throws IllegalArgumentException Invalid ID
      */
-    public static @Nonnull String getImageFingerprintHash(@Nonnull String imageId) {
-        if (imageId.length() != 64) {
-            throw new IllegalArgumentException("Expecting 64-char full image ID, but got " + imageId);
+    public static @Nonnull String getFingerprintHash(@Nonnull String id) {
+        if (id.length() != 64) {
+            throw new IllegalArgumentException("Expecting 64-char full image ID, but got " + id);
         }
-        return imageId.substring(0, 32);
+        return id.substring(0, 32);
     }
 
     /**
-     * Gets {@link Fingerprint} for a given docker image.
+     * Gets {@link Fingerprint} for a given docker ID.
+     * @param id Docker ID (image or container). Only 64-char full IDs are supported.
+     * @return Created fingerprint or null if it is not found
+     * @throws IOException Fingerprint loading error
      */
-    public static @CheckForNull Fingerprint of(@Nonnull String imageId) throws IOException {
-        return Jenkins.getInstance().getFingerprintMap().get(getImageFingerprintHash(imageId));
+    public static @CheckForNull Fingerprint of(@Nonnull String id) throws IOException {
+        return Jenkins.getInstance().getFingerprintMap().get(getFingerprintHash(id));
     }
-
-    private static @Nonnull Fingerprint make(@Nonnull Run<?,?> run, @Nonnull String imageId) throws IOException {
-        return Jenkins.getInstance().getFingerprintMap().getOrCreate(run, "<docker-image>", getImageFingerprintHash(imageId));
+    
+    private static @CheckForNull Fingerprint ofNoException(@Nonnull String id) {
+        try {
+            return of(id);
+        } catch (IOException ex) { // The error is not a hazard in CheckForNull logic
+            LOGGER.log(Level.WARNING, "Cannot retrieve a fingerprint for Docker id="+id, ex);
```

Comment:
Should this really be `WARNING`? Seems like it is fine to ask about a nonexistent ID, in which case this should be `FINE`.